frontend/src/app/supportAdmin/UnarchivePatients/UnarchivePatient.tsx

```diff
@@ -0,0 +1,425 @@
+  patients: UnarchivePatientPatient[] | undefined,
+  patientsCount: number | undefined
+) => {
+  return patients !== undefined && patientsCount !== undefined;
+};
+
+const displayPagination = (
+  patients: UnarchivePatientPatient[] | undefined,
+  patientsCount: number | undefined
+) => {
+  if (patients && patientsCount) {
+    return patientsCount > 0 && patients.length > 0;
+  } else {
+    return false;
+  }
+};
+
+const displayInstructions = (
+  selectedOrgId: string,
+  patients: UnarchivePatientPatient[] | undefined,
+  patientsCount: number | undefined,
+  facilities: UnarchivePatientFacility[]
+) => {
+  if (
+    !displayPatientTable(patients, patientsCount) &&
+    !displayNoFacilitiesMsg(selectedOrgId, facilities)
+  ) {
+    return (
+      <div className="width-full margin-top-2">
+        Filter by organization and testing facility to display archived
+        patients.
+      </div>
+    );
+  }
+};
+
+const displayNoFacilitiesMsg = (
+  selectedOrgId: string,
+  facilities: UnarchivePatientFacility[]
+) => {
+  if (selectedOrgId && facilities?.length === 0) {
+    return (
+      <div className="width-full margin-top-2">
+        This organization has no facilities. Select a different organization.
+      </div>
+    );
+  }
+};
+
+const getOrgExternalIdWithInternalId = (
+  orgs: UnarchivePatientOrganization[],
+  internalId: string
+) => {
+  let org = orgs.find((org) => org.internalId === internalId);
+  return org?.externalId;
+};
+const UnarchivePatient = ({
+  organizations,
+  currentPage,
+  selectedFacilityId,
+  setSelectedFacilityId,
+  selectedOrgId,
+  setSelectedOrgId,
+  archivedPatientsCount,
+  setArchivedPatientsCount,
+  facilities,
+  setFacilities,
+  archivedPatients,
+  setArchivedPatients,
+}: UnarchivePatientProps) => {
+  const pageTitle = `Unarchive ${PATIENT_TERM}`;
+  useDocumentTitle(pageTitle);
+  const unarchivePatientPageUrl = "/admin/unarchive-patient";
+  const entriesPerPage = 20;
+
+  const [isLoadingCount, setIsLoadingCount] = useState(false);
+  const [isLoadingPatients, setIsLoadingPatients] = useState(false);
+
+  const [activeFacility] = useSelectedFacility();
+  const navigate = useNavigate();
+
+  const [queryGetOrgWithFacilities] = useGetOrganizationWithFacilitiesLazyQuery(
+    {
+      fetchPolicy: "no-cache",
+    }
+  );
+  const [queryGetPatientsByFacilityWithOrg] =
+    useGetPatientsByFacilityWithOrgLazyQuery({
+      fetchPolicy: "no-cache",
+    });
+
+  const [queryGetPatientsCountByFacilityWithOrg] =
+    useGetPatientsCountByFacilityWithOrgLazyQuery({
+      fetchPolicy: "no-cache",
+    });
+
+  const {
+    getValues,
+    handleSubmit,
+    register,
+    watch,
+    formState: { errors },
+  } = useForm({});
+
+  const formCurrentValues = watch();
+
+  useEffect(() => {
+    fetchAndSetArchivedPatients(
+      selectedOrgId,
+      selectedFacilityId,
+      organizations
+    );
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [currentPage]);
```

Comment:
⚠️ I'm using `useEffect` here to update the patients when a user interacts with the pagination

Reply:
Could you help me understand the decision behind using `useEffect` here with `currentPage` as the dependency to build in a "subscription" to the pagination changing rather than attaching an equivalent `onChange` handler to the pagination component? I think the useEffect binding here is what's causing the behavior described [here](https://github.com/CDCgov/prime-simplereport/pull/6299#discussion_r1290399247) since the component root is re-rendering when the currentPage value is being changed (in that case manually). 

It may be cleaner / more idiomatic to define an onPaginationChange function here and then pass it down rather than handling it in a useEffect like this. Not sure what the lift is like to do something like this though (given that this is working so far) so lmk what the tradeoffs look like!

Reply:
Thanks for pointing this out, Bob! I changed up the `Pagination` component a bit to avoid using the `useEffect` hook as I was before. Let me know what you think!